lib/ansible/modules/network/nxos/nxos_reboot.py

```diff
@@ -86,7 +87,7 @@ def main():
     changed = True
 
     results = {
-        'changed': True,
+        'changed': changed,
```

Comment:
Can you remove this block of code [L89-L92](https://github.com/Akasurde/ansible/blob/fd1e79d33e03ee10e78f512f04954ae29e9a451e/lib/ansible/modules/network/nxos/nxos_reboot.py#L89-L92)? And we can have `results` initialized like `dict(changed=False, warnings=warnings)` instead?
In that case `changed = True` should also be updated to `results['changed'] = True`.